lib/services/project-data-service.ts

```diff
@@ -51,6 +53,137 @@ export class ProjectDataService implements IProjectDataService {
+		const hasMigrated = this.$androidResourcesMigrationService.hasMigrated(projectData.appResourcesDirectoryPath);
+		const basePath = hasMigrated ? path.join(pathToAndroidDir, SRC_DIR, MAIN_DIR, RESOURCES_DIR) : pathToAndroidDir;
+
+		const currentStructure = this.$fs.enumerateFilesInDirectorySync(basePath);
+		const content = this.getImageDefinitions().android;
+
+		return {
+			icons: this.getAndroidAssetSubGroup(content.icons, currentStructure),
+			splashBackgrounds: this.getAndroidAssetSubGroup(content.splashBackgrounds, currentStructure),
+			splashCenterImages: this.getAndroidAssetSubGroup(content.splashCenterImages, currentStructure),
+			splashImages: null
+		};
+	}
+
+	private getImageDefinitions(): any {
+		const pathToImageDefinitions = path.join(__dirname, "..", "..", CLI_RESOURCES_DIR_NAME, AssetConstants.assets, AssetConstants.imageDefinitionsFileName);
+		const imageDefinitions = this.$fs.readJson(pathToImageDefinitions);
+
+		return imageDefinitions;
+	}
+
+	private async getIOSAssetSubGroup(dirPath: string): Promise<IAssetSubGroup> {
+		const pathToContentJson = path.join(dirPath, AssetConstants.iOSResourcesFileName);
+		const content = <IAssetSubGroup>this.$fs.readJson(pathToContentJson);
+
+		const imageDefinitions = this.getImageDefinitions().ios;
+
+		if (content && content.images) {
+			_.each(content.images, image => {
+				// In some cases the image may not be available, it will just be described.
+				// When this happens, the filename will be empty.
+				// So we'll keep the path empty as well.
+				if (image.filename) {
+					image.path = path.join(dirPath, image.filename);
+				}
+
+				if (image.size) {
+					// size is basically <width>x<height>
+					const [width, height] = image.size.toString().split(AssetConstants.sizeDelimiter);
+					if (width && height) {
+						image.width = +width;
+						image.height = +height;
+					}
+				} else {
+					// Find the image size based on the hardcoded values in the image-definitions.json
+					_.each(imageDefinitions, (assetSubGroup: IAssetItem[]) => {
+						_.each(assetSubGroup, assetItem => {
+							if (assetItem.filename === image.filename && path.basename(assetItem.directory) === path.basename(dirPath)) {
+								image.width = assetItem.width;
+								image.height = assetItem.height;
+								image.size = `${assetItem.width}${AssetConstants.sizeDelimiter}${assetItem.height}`;
+							}
+						});
+					});
+				}
+			});
+		}
+
+		return content;
+	}
+
+	private getAndroidAssetSubGroup(assetItems: IAssetItem[], realPaths: string[]): IAssetSubGroup {
+		const assetSubGroup: IAssetSubGroup = {
+			images: <any>[]
+		};
+
+		const normalizedPaths = _.map(realPaths, p => path.normalize(p));
+		_.each(assetItems, assetItem => {
+			_.each(normalizedPaths, currentNormalizedPath => {
+				const imagePath = path.join(assetItem.directory, assetItem.filename);
+				if (currentNormalizedPath.indexOf(path.normalize(imagePath)) !== -1) {
```

Comment:
You can use `_.find` to locate the value of interest instead of using `_.each` and then `return false` (e.g. `break`)

Reply:
I've used find at the beginning, but it looks strange, so I prefer the each